test/fast_reader_selftest.c

```
@@ -0,0 +1,196 @@
+        free(gzb);
+    }
+
+    /* --- multi-member gzip (cat a.gz b.gz) --- */
+    {
+        size_t half = n / 2;
+        size_t cap = n + (n >> 1) + 8192;
+        unsigned char *a = malloc(cap), *b = malloc(cap);
+        size_t la = gz_member(pay, half, a, cap);
+        size_t lb = gz_member(pay + half, n - half, b, cap);
+        unsigned char *cat = malloc(la + lb);
+        memcpy(cat, a, la); memcpy(cat + la, b, lb);
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)